nav2_behavior_tree/plugins/decorator/path_longer_on_approach.cpp

```diff
@@ -34,9 +34,9 @@ bool PathLongerOnApproach::isPathUpdated(
   nav_msgs::msg::Path & new_path,
   nav_msgs::msg::Path & old_path)
 {
-  return new_path != old_path && old_path.poses.size() != 0 &&
+  return old_path.poses.size() != 0 &&
          new_path.poses.size() != 0 &&
-         old_path.poses.back().pose == new_path.poses.back().pose;
+         old_path.poses.back().pose.position == new_path.poses.back().pose.position;
```

Comment:
Wait, what happened to checking the sizes are the same `new_path.poses.size() == old_path.poses.size()`?

Reply:
Oh yeahh.. you mentioned that in you previous comments. 

The function checks `isPathUpdated`, having the equality means then the function returns true, when the path is not updated. Therefore it should be `!=`. Will update it soon..

Reply:
Sorry it's taking longer than expected. I'd be on it soon 🥲